animations/lib/src/basics/08_animated_list.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+class AnimatedListDemo extends StatefulWidget {
+  static String routeName = '/basics/08_animated_list';
+
+  _AnimatedListDemoState createState() => _AnimatedListDemoState();
+}
+
+class _AnimatedListDemoState extends State<AnimatedListDemo> {
+  final GlobalKey<AnimatedListState> _listKey = GlobalKey();
+  List<UserModel> initialListData = listData;
+
+  void addUser() {
+    setState(() {
+      int index = initialListData.length;
+      _listKey.currentState
+          .insertItem(index, duration: Duration(milliseconds: 300));
+      initialListData.add(
```

Comment:
I would move this statement to be above the call to `insertItem`. Better to update the list before you tell the AnimatedListView to begin animating.